In [1]:
import pandas as pd
data = pd.read_csv("thanksgiving.csv", encoding="Latin-1")
data.head(1)

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            

        ...         \
0       ...          

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            

  Have you ever attended a "Friendsgiving?"  \
0                                        No   

  Will you shop any Black Friday sales on Thanksgiving Day?  \
0                                                 No          

  Do you work in retail? Will you employer make you work on Black Friday?  \
0                     No                                              NaN   

  How would you describe where you live?      Age What is your gender?  \
0                               Suburban  18 - 29                 Male   

  How much total combined money did all members of your HOUSEHOLD earn last year?  \
0                                 $75,000 to $99,999                                

         US Region  
0  Middle Atlantic  

[1 rows x 65 columns]

In [2]:
data.columns

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

In [3]:
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [4]:
boolean = (data['Do you celebrate Thanksgiving?'] == 'Yes')
data = data.loc[boolean]
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
Name: Do you celebrate Thanksgiving?, dtype: int64

In [5]:
data['What is typically the main dish at your Thanksgiving dinner?'].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [6]:
filt_tofur = data[data['What is typically the main dish at your Thanksgiving dinner?'] == 'Tofurkey']
filt_tofur['Do you typically have gravy?']

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

In [7]:
apple_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple']) 
pumpkin_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin'])
pecan_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan'])
ate_pies = apple_isnull & pumpkin_isnull & pecan_isnull
ate_pies.value_counts()

False    876
True     104
dtype: int64

In [8]:
print(data['Age'].value_counts())

45 - 59    269
60+        258
30 - 44    235
18 - 29    185
Name: Age, dtype: int64


In [9]:
def convert(string):
    if pd.isnull(string) == True:
        return None
    else:
        string = string.split(' ')[0]
        answer = string.replace('+', '')
        return int(answer)

data['int_age'] = data["Age"].apply(convert)
data['int_age'].describe()

count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dtype: float64

Despite the fact that we have rounded off the age, the data reflects the present proportions of age groups (the same proportions as in [9])

In [10]:
def income(string):
    if pd.isnull(string) == True:
        return None
    else:
        string = string.split(' ')[0]
        if string == 'Prefer':
            return None
        string = string.replace('$', '')
        string = string.replace(',', '')
        return int(string)
data['int_income'] = data['How much total combined money did all members of your HOUSEHOLD earn last year?'].apply(income)
data['int_income'].describe()

count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64

In [11]:
print(data['How much total combined money did all members of your HOUSEHOLD earn last year?'].value_counts())

$25,000 to $49,999      166
$50,000 to $74,999      127
$75,000 to $99,999      127
Prefer not to answer    118
$100,000 to $124,999    109
$200,000 and up          76
$10,000 to $24,999       60
$0 to $9,999             52
$125,000 to $149,999     48
$150,000 to $174,999     38
$175,000 to $199,999     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64


When analyzing this result, it should be borne in mind that we took only the lower value of each group. Also, we have a very high standard deviation (59 thousand dollars) with the step size of each group of 25 thousand.

In [12]:
data[data['int_income'] < 50000]['How far will you travel for Thanksgiving?'].value_counts(normalize=True)

Thanksgiving is happening at my home--I won't travel at all                         0.381295
Thanksgiving is local--it will take place in the town I live in                     0.330935
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    0.230216
Thanksgiving is out of town and far away--I have to drive several hours or fly      0.057554
Name: How far will you travel for Thanksgiving?, dtype: float64

In [13]:
data[data['int_income'] >= 150000]['How far will you travel for Thanksgiving?'].value_counts(normalize=True)

Thanksgiving is happening at my home--I won't travel at all                         0.471429
Thanksgiving is local--it will take place in the town I live in                     0.242857
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    0.178571
Thanksgiving is out of town and far away--I have to drive several hours or fly      0.107143
Name: How far will you travel for Thanksgiving?, dtype: float64

Only 38 percent of the respondents (with an income of 50000 thousand) celebrate at home. At the same time, 47 percent of respondents (with an income of 150 and above) celebrate at home. Despite not much difference, we can assume that students prefer to celebrate a holiday in the home of parents who earn more (and accordingly, celebrate in their own home).

In [15]:
table = pd.pivot_table(data, values='int_age', index='Have you ever tried to meet up with hometown friends on Thanksgiving night?', columns='Have you ever attended a "Friendsgiving?"')
table

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [16]:
table_income = pd.pivot_table(data, values='int_income', index='Have you ever tried to meet up with hometown friends on Thanksgiving night?', columns='Have you ever attended a "Friendsgiving?"')
table_income

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

It appears that young people prefer to meet up with friends on Thanksgiving and try to attend a "Friendsgiving? (We can make this conclusion  from pivot tables: people who do this have less age (33.9) and income(66019))

## Figure out the most common dessert people eat.

In [18]:
none1 = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Other (please specify)'].value_counts()
none2 = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Other (please specify).1'].value_counts()

In [19]:
none1

Other (please specify)    134
Name: Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Other (please specify), dtype: int64

In [20]:
none2

pie                                                    13
Pie                                                    12
pies                                                    6
Pumpkin pie                                             4
pumpkin pie                                             4
chocolate cake                                          3
Pies                                                    3
cake                                                    2
Pumpkin Pie                                             2
Flan                                                    2
pie-spple & pecan                                       1
Cake (not carrot)                                       1
banana pudding                                          1
Other bars                                              1
pie is the dessert                                      1
pistachio ambrosia salad                                1
Date balls                                              1
German Chocola

In [21]:
data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Peach cobbler'].value_counts()

Peach cobbler    103
Name: Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Peach cobbler, dtype: int64